components/brave_wallet/browser/brave_wallet_pin_service.cc

```diff
@@ -507,20 +595,56 @@ void BraveWalletPinService::OnTokenMetaDataReceived(
   std::vector<std::string> cids;
 
   cids.push_back(ExtractCID(token_url).value());
-  auto* image = parsed_result->FindStringKey("image");
-  if (image) {
-    auto image_cid = ExtractCID(*image);
+  auto* image_url = parsed_result->FindStringKey("image");
+  if (image_url) {
+    auto image_cid = ExtractCID(*image_url);
     if (image_cid) {
-      cids.push_back(image_cid.value());
+      cids.push_back(*image_cid);
+      content_type_checker_->CheckContentTypeSupported(
+          *image_url,
+          base::BindOnce(&BraveWalletPinService::OnContentTypeChecked,
+                         weak_ptr_factory_.GetWeakPtr(), std::move(service),
+                         std::move(token), std::move(cids),
+                         std::move(callback)));
```

Comment:
This needs S&P review as we are getting arbitrary url resource from blockchain and forcing browser to make a request to it.

Reply:
It is a bit different : actually we make a call to the IPFS gateway which we use to fetch metadata on the previous step. Final url looks like https://ipfs.io/CID/...

Reply:
I think it's fine. It's not arbitrary, like @cypt4 said and we only call IPFS gateway. This code also doesn't fetch a file, only response headers.